src/Fixer/Comment/SingleLineCommentStyleFixer.php

```diff
@@ -122,6 +122,11 @@ protected function applyFix(\SplFileInfo $file, Tokens $tokens)
 
                 continue;
             }
+
+            if (false !== strpos($commentContent, '?>')) {
```

Comment:
I would merge this in the check below, after the "cheaper" checks (like `!$this->asteriskEnabled`),
for reference: https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/3320/files